src/components/LayoutTree.tsx

```diff
@@ -426,14 +560,39 @@ const TreeNode = ({
   cliffPaddingStyle: { paddingBottom: number }
   dragInProgress: boolean
   autofocusDepth: number
+  cursorIsGreaterThanPrevSibling: boolean
 } & Pick<CSSTransitionProps, 'in'>) => {
   const [y, setY] = useState(_y)
+  const [x, setX] = useState(_x)
   const fadeThoughtRef = useRef<HTMLDivElement>(null)
+
+  const isParentSwapped = useSelector(state => {
+    const allPatches = state.undoPatches
+    return allPatches?.flat().some(patch => patch.actions[0] === 'swapParent')
+  })
+
+  // true if the last action is newThought
   const isLastActionNewThought = useSelector(state => {
     const lastPatches = state.undoPatches[state.undoPatches.length - 1]
     return lastPatches?.some(patch => patch.actions[0] === 'newThought')
   })
 
+  // true if the last action is swapParent
+  const isLastActionSwapParent = useSelector(state => {
+    const lastPatches = state.undoPatches[state.undoPatches.length - 1]
+    return lastPatches?.some(patch => patch.actions[0] === 'swapParent')
+  })
+
+  const isLastActionEditThought = useSelector(state => {
+    const lastPatches = state.undoPatches[state.undoPatches.length - 1]
+    return lastPatches?.some(patch => patch.actions[0] === 'editThought')
+  })
```

Comment:
This can't be right. Nowhere in the intended behavior does it say that `swapParent` should behave differently if the last action was `editThought`.

Reply:
Please see this comment: https://github.com/cybersemics/em/pull/2810#discussion_r2030008163

Reply:
I've fixed the approach in https://github.com/cybersemics/em/pull/2810/commits/a3fd82890102922d18a4bf362b590f2ba2fa18e1